pkg/manager/internal.go

```diff
@@ -70,8 +81,15 @@ type controllerManager struct {
 	errChan              chan error
 	runnables            *runnables
 
-	// cluster holds a variety of methods to interact with a cluster. Required.
-	cluster cluster.Cluster
+	// defaultCluster holds a variety of methods to interact with a defaultCluster. Required.
+	defaultCluster        cluster.Cluster
+	defaultClusterOptions cluster.Option
+
+	logicalProvider cluster.Provider
```

Comment:
```suggestion
	clusterProvider cluster.Provider
```
nit: Similarly we probably don't need logical as prefix in l.90-92

Reply:
done